src/System.Threading.AccessControl/tests/EventWaitHandleAclTests.cs

```diff
@@ -0,0 +1,269 @@
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.FullControl, AccessControlType.Allow)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.FullControl, AccessControlType.Deny)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.Synchronize, AccessControlType.Allow)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.Synchronize, AccessControlType.Deny)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.Modify,      AccessControlType.Allow)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.Modify,      AccessControlType.Deny)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize, AccessControlType.Allow)]
+        [InlineData(true, EventResetMode.ManualReset,  EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize, AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.FullControl, AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.FullControl, AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.Synchronize, AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.Synchronize, AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.Modify,      AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.Modify,      AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize, AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.AutoReset,   EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize, AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.FullControl, AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.FullControl, AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.Synchronize, AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.Synchronize, AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.Modify,      AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.Modify,      AccessControlType.Deny)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize, AccessControlType.Allow)]
+        [InlineData(false, EventResetMode.ManualReset, EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize, AccessControlType.Deny)]
```

Comment:
If you're trying to create a matrix of inputs, it'd be cleaner and easier to understand if you use a MemberData, e.g.
```C#
public static IEnumerable<object[]> EventWaitHandle_Create_SpecificParameters_MemberData() =>
    from initialState in new[] { true, false }
    from mode in new[] { EventResetMode.AutoReset, EventResetMode.ManualReset }
    from rights in new[] { EventWaitHandleRights.FullControl, EventWaitHandleRights.Synchronize, EventWaitHandleRights.Modify, EventWaitHandleRights.Modify | EventWaitHandleRights.Synchronize }
    from accessControl in new[] { AccessControlType.Allow, AccessControlType.Deny }
    select new object[] { initialState, mode, rights accessControl };

[Theory]
[MemberData(nameof(EventWaitHandle_Create_SpecificParameters_MemberData))]
public void EventWaitHandle_Create_SpecificParameters(bool initialState, EventResetMode mode, EventWaitHandleRights rights, AccessControlType accessControl)
{
    ...
}
```

Reply:
Nice! Didn't know about that option. Thanks for letting me know. Fixing it.

Reply:
That is a neat approach!